### PR TITLE
Fix immutable tests

### PR DESCRIPTION
**Pull request by gianbelinche**
This PR fixes the tests under `solidity/simple/immutable`

### CODE REVIEW

**Review by jorbush — APPROVED** [review]
Nice work! 
Also fixes: 
- `tests/solidity/simple/expression`
- `tests/solidity/simple/gas_value`

File: src/address_operands.rs [hunk 1]
```diff
@@ -176,11 +176,7 @@ fn address_operands(
                 ImmMemHandlerFlags::UseStackWithPushPop => {
                     // stack+=[src0 + offset] + src1
                     let src0 = reg_and_imm_write(vm, OutputOperandPosition::First, opcode);
-                    vm.current_frame_mut()?.sp += src0.value.low_u64() + 1;
-                    let sp = vm.current_frame()?.sp;
-                    vm.current_context_mut()?
-                        .stack
-                        .store_with_offset(1, res.0, sp)?;
+                    vm.current_frame_mut()?.sp += src0.value.low_u64();
```
Comment (Oppen) [inline src/address_operands.rs:179]: Stacks are 16-bits addressed, so I'm not sure `low_u64`  is enough bits dropped.

File: src/op_handlers/far_call.rs [hunk 2]
```diff
@@ -262,7 +262,6 @@ pub fn far_call(
 
             // set calldata pointer
             vm.set_register(1, TaggedValue::new_pointer(forward_memory.encode()));
-            vm.current_context_mut()?.caller = address_from_u256(&vm.get_register(15).value);
```
Comment (Oppen) [inline src/op_handlers/far_call.rs:265]: I don't exactly understand this bit. Is this due to static mode being in effect?

Comment (gianbelinche) [reply src/op_handlers/far_call.rs:265]: No, in mimic calls the caller is set to the value of register 15

On the formal spec:
```
Mimic call sets:
this_address <- destination address;
msg_sender <- value of r15;
context <- value of context register gs_context_u128
```

File: src/address_operands.rs (hunk 1 above)
Comment (gianbelinche) [reply src/address_operands.rs]: Done

**Review by juan518munoz — APPROVED** [review]
LGTM

**Review by Oppen — APPROVED** [review]

**Pull request merged**